### PR TITLE
Userland: Replacing DeprecatedString [2048, Chess]

### PR DESCRIPTION
**Pull request by JRedrupp**
Raising these migrations from DeprecatedString in order to assist with issue https://github.com/SerenityOS/serenity/issues/17128
No new Features added, purely a migration.

Ran into some issues with the rebase, and had to create a new PR (including the requested updates).
Previous PR: #19032

### CODE REVIEW

File: Userland/Games/Chess/ChessWidget.cpp [hunk 1]
```diff
@@ -548,12 +548,12 @@ ErrorOr<void> ChessWidget::import_pgn(Core::File& file)
     bool recursive_annotation = false;
     bool future_expansion = false;
     Chess::Color turn = Chess::Color::White;
-    DeprecatedString movetext;
+    String movetext;
 
     for (size_t j = i; j < lines.size(); j++)
-        movetext = DeprecatedString::formatted("{}{}", movetext, lines.at(i).to_deprecated_string());
+        movetext = String::formatted("{}{}", movetext, lines.at(i)).release_value_but_fixme_should_propagate_errors();
```
Comment (Dexesttp) [inline Userland/Games/Chess/ChessWidget.cpp:554]: I think this can be `TRY()`'d, since the method is `ErrorOr<void>`?

Comment (JRedrupp) [reply Userland/Games/Chess/ChessWidget.cpp]: @Dexesttp - Thanks for the review - Have just resolved the above

File: Userland/Games/Chess/ChessWidget.cpp [hunk 2]
```diff
@@ -522,9 +522,9 @@ void ChessWidget::playback_move(PlaybackDirection direction)
     update();
 }
 
-DeprecatedString ChessWidget::get_fen() const
+String ChessWidget::get_fen() const
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.cpp:525]: I don't think that this is a step in the correct direction.
If a function will ever return a normal `String` (I mean no SSO involved), we should return a `ErrorOr<String>`, not a `String`.

I think that we should properly convert this function and add the `.release_value_but_fixme_should_propagate_errors()` on the caller side.

File: Userland/Games/Chess/ChessWidget.cpp [hunk 3]
```diff
@@ -629,16 +629,16 @@ ErrorOr<void> ChessWidget::export_pgn(Core::File& file) const
     // Tag Pair Section
     TRY(file.write_until_depleted("[Event \"Casual Game\"]\n"sv.bytes()));
     TRY(file.write_until_depleted("[Site \"SerenityOS Chess\"]\n"sv.bytes()));
-    TRY(file.write_until_depleted(DeprecatedString::formatted("[Date \"{}\"]\n", Core::DateTime::now().to_deprecated_string("%Y.%m.%d"sv)).bytes()));
+    TRY(file.write_until_depleted(TRY(String::formatted("[Date \"{}\"]\n", TRY(Core::DateTime::now().to_string("%Y.%m.%d"sv)))).bytes()));
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.cpp:632]: Using `write_formatted` would be easier here (and also many times later).

File: Userland/Games/Chess/ChessWidget.h [hunk 4]
```diff
@@ -43,7 +43,7 @@ class ChessWidget final
     void set_side(Chess::Color side) { m_side = side; };
 
     void set_piece_set(StringView set);
-    DeprecatedString const& piece_set() const { return m_piece_set; };
+    StringView const& piece_set() const { return m_piece_set; };
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.h:46]: It looks like this function isn't used, so we should probably get rid of it instead.
And note that providing an API to access data that we don't own is probably a bit silly.

File: Userland/Games/2048/BoardView.cpp [hunk 5]
```diff
@@ -234,7 +234,7 @@ void BoardView::paint_event(GUI::PaintEvent& event)
             auto rect = Gfx::IntRect::centered_on(center, tile_size);
 
             painter.fill_rect(rect, background_color_for_cell(sliding_tile.value_from));
-            painter.draw_text(rect, DeprecatedString::number(sliding_tile.value_from), font(), Gfx::TextAlignment::Center, text_color_for_cell(sliding_tile.value_from));
```
Comment (LucasChollet) [inline Userland/Games/2048/BoardView.cpp:237]: Looks like `, font()` unintentionally disappeared here.

File: Userland/Games/Chess/ChessWidget.h [hunk 6]
```diff
@@ -140,7 +139,7 @@ class ChessWidget final
     Color m_marking_secondary_color { Color::from_argb(0x6655dd55) };
     Chess::Color m_side { Chess::Color::White };
     HashMap<Chess::Piece, RefPtr<Gfx::Bitmap const>> m_pieces;
-    DeprecatedString m_piece_set;
+    StringView m_piece_set;
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.h:142]: I wasn't very clear, but my comment is also valid for this `StringView`. Keeping a `StringView` to some data that you don't own can be quite silly.
This field isn't used anywhere, so you can delete it. Btw that can be in its own commit.

File: Userland/Games/Chess/ChessWidget.cpp [hunk 7]
```diff
@@ -648,13 +648,13 @@ ErrorOr<void> ChessWidget::export_pgn(Core::File& file) const
 
     // Movetext Section
     for (size_t i = 0, move_no = 1; i < m_board.moves().size(); i += 2, move_no++) {
-        const DeprecatedString white = m_board.moves().at(i).to_algebraic().release_value_but_fixme_should_propagate_errors().to_deprecated_string();
+        const String white = TRY(m_board.moves().at(i).to_algebraic());
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.cpp:651]: We try to use east-const now, can you please move this one?

File: Userland/Games/Chess/ChessWidget.cpp [hunk 8]
```diff
@@ -648,13 +648,13 @@ ErrorOr<void> ChessWidget::export_pgn(Core::File& file) const
...
 
         if (i + 1 < m_board.moves().size()) {
-            const DeprecatedString black = m_board.moves().at(i + 1).to_algebraic().release_value_but_fixme_should_propagate_errors().to_deprecated_string();
-            TRY(file.write_until_depleted(DeprecatedString::formatted("{}. {} {} ", move_no, white, black).bytes()));
+            const String black = TRY(m_board.moves().at(i + 1).to_algebraic());
```
Comment (LucasChollet) [inline Userland/Games/Chess/ChessWidget.cpp:654]: And here
